src/kernels/MIOpenInterpolate.cpp

```diff
@@ -0,0 +1,1057 @@
+                                                       uint64_t Hout,
+                                                       bool align_corners,
+                                                       uint64_t* hin_index0,
+                                                       uint64_t* hin_index1,
+                                                       FLOAT_ACCUM* lambda0,
+                                                       FLOAT_ACCUM* lambda1)
+{
+    FLOAT_ACCUM hin_index_actual = max(0., get_src_index(h, scale_factor, align_corners));
+    *hin_index0                  = static_cast<uint64_t>(hin_index_actual);
+    *hin_index1                  = min(*hin_index0 + 1, Hin - 1);
+    *lambda1                     = hin_index_actual - *hin_index0;
+    *lambda0                     = 1.f - *lambda1;
+}
+
+__device__ inline FLOAT_ACCUM get_back_lambda(
+    uint64_t src, uint64_t src0, uint64_t src1, FLOAT_ACCUM lambda0, FLOAT_ACCUM lambda1)
+{
+    if(src == src0)
+    {
+        if(src0 == src1)
+        {
+            return 1; // lambda0 + lambda1 = 1
+        }
+        return lambda0;
+    }
+    if(src == src1)
+    {
+        return lambda1;
+    }
+    // This case can happen due to floating point mutiplification.
+    // ex> 7 * (105/9) = 87 or 86.99999995
+    return 0;
+}
+
+__device__ inline FLOAT_ACCUM compute_back_lambda(uint64_t dest,
+                                                  uint64_t src,
+                                                  FLOAT_ACCUM scale_factor,
+                                                  uint64_t Hin,
+                                                  uint64_t Hout,
+                                                  bool align_corners)
+{
+    if(Hin == Hout)
+    {
+        return 1;
+    }
+    uint64_t index0;
+    uint64_t index1;
+    FLOAT_ACCUM lambda0;
+    FLOAT_ACCUM lambda1;
+    compute_source_index_and_lambda(
+        dest, scale_factor, Hin, Hout, align_corners, &index0, &index1, &lambda0, &lambda1);
+    return get_back_lambda(src, index0, index1, lambda0, lambda1);
+}
+
+template <typename TI, typename TO>
+__device__ inline void interpolateLinearForward(const TI* __restrict__ input,
+                                                TO* __restrict__ output,
+                                                const tensor_view_t<3> input_tv,
+                                                const tensor_view_t<3> output_tv,
+                                                const size_t nelems,
+                                                const float* scale_factors,
+                                                const bool align_corners)
+{
+    uint64_t gid = blockIdx.x * blockDim.x + threadIdx.x;
+    if(gid >= nelems)
+        return;
+
+    auto tensor_layout = tensor_layout_t<3>(output_tv, gid);
+    uint64_t n         = tensor_layout.layout[0];
+    uint64_t c         = tensor_layout.layout[1];
+    uint64_t h         = tensor_layout.layout[2];
```

Comment:
You can take a look at this PR: https://github.com/moreh-dev/MIOpen/pull/37. I have made some changes to the tensor_view.hpp and I think you can implement it here.

Reply:
changed as per your recommend